src/services/AudioMixer.cxx

```diff
@@ -367,11 +366,11 @@ void AudioMixer::playSoundtrackWithEffect(SoundtrackUPtr& track, int ALEffect)
   alDeleteEffects(1, &effect);
 
   //apply effect to source
-  alSource3i(track->source, AL_AUXILIARY_SEND_FILTER, (ALint)(track->effect_slot), 0, AL_FILTER_NULL);
-  assert(alGetError() == AL_NO_ERROR && "Failed to setup effect for sound source send 0.");
+  alSource3i(track->source[track->isMusic], AL_AUXILIARY_SEND_FILTER, (ALint)(track->effect_slot), 0, AL_FILTER_NULL);
+  assert(alGetError() == AL_NO_ERROR && "Failed to setup reverb for sound source send 0.");
 
   //play sound
-  alSourcePlay(track->source);
+  alSourcePlay(track->source[track->isMusic]);
```

Comment:
same as before

Reply:
The reason why I did this is to cut down on unecessary lines because if I did a hardcoded value of 0 or 1 (or even if I implement an enum with something like `SFX` and `MUSIC`) instead of having just one line like:
```c++
alSourcePlay(track->source[track->isMusic]);
```
I would need to have at least 3 lines to do the same thing:
```c++
if(track->isMusic])
  alSourcePlay(track->source[MUSIC]);
else
  alSourcePlay(track->source[SFX]);
```

And I feel like that's so much more convoluted and error prone. So that was my rationale behind it, I don't know how you feel about that.

Reply:
ok, accepted

Reply:
Do you still want me to try and do the enum if it will add those extra if statements?

Reply:
no I think it's ok
since you have a comment